indexers/keyvalue/PostgreSQLDBMSIndexer/__init__.py

```diff
@@ -0,0 +1,144 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import pickle
+from typing import Optional
+
+from jina.executors.indexers import BaseIndexer
+
+
+class PostgreSQLDBMSIndexer(BaseIndexer):
+    """:class:`PostgreSQLDBMSIndexer` PostgreSQL based KV Indexer.
+        Initialize the PostgreSQLDBIndexer.
+
+        :param hostname: hostname of the machine
+        :param port: the port
+        :param username: the username to authenticate
+        :param password: the password to authenticate
+        :param database: the database name
+        :param collection: the collection name
+        :param args: other arguments
+        :param kwargs: other keyword arguments
+    """
+
+    def __init__(self,
+                 hostname: str = '127.0.0.1',
+                 port: int = 5432,
+                 username: str = 'default_name',
+                 password: str = 'default_pwd',
+                 database: str = 'default_db',
+                 table: Optional[str] = 'default_table',
+                 *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        self.hostname = hostname
+        self.port = port
+        self.username = username
+        self.password = password
+        self.database_name = database
+        self.table = table
+
+    def __enter__(self):
```

Comment:
AFAIK, no other db indexer is a context manager. This should just be a `get_query_handler` . See https://github.com/jina-ai/jina-hub/blob/70cc924619aa7b350d0b26a960ed08c1ca5299a8/indexers/keyvalue/RedisDBIndexer/__init__.py#L28